client/core/trade.go

```diff
@@ -1438,83 +1437,102 @@ func (c *Core) swapMatchGroup(t *trackedTrade, matches []*matchTracker, errs *er
 		if secret := match.MetaData.Proof.Secret; len(secret) > 0 {
 			c.log.Tracef("Contract coin %v secret = %x", coin, secret)
 		}
-		err := c.finalizeSwapAction(t, match, coin.ID(), receipt.Contract())
-		if err != nil {
-			errs.addErr(err)
+
+		// Update the match db data with the swap details before attempting
+		// to notify the server of the swap.
+		proof := &match.MetaData.Proof
+		contract, coinID := receipt.Contract(), []byte(coin.ID())
+		proof.Script = contract
+		if match.Match.Side == order.Taker {
+			proof.TakerSwap = coinID
+			match.SetStatus(order.TakerSwapCast)
+		} else {
+			proof.MakerSwap = coinID
+			match.SetStatus(order.MakerSwapCast)
```

Comment:
As opposed to my comment on setting status to `MatchComplete` before the `redeem`, I think this is fine as you have it here before the `init`. We are running `resendPendingRequests` every tick and have plenty of time to get the server updated.